linchpin/provision/roles/azure/tasks/provision_azure_loadbalancer.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Set Default Variables
+  block:
+  - set_fact:
+      inbound_nat_rules: ["{{ res_def['inbound_nat_rules'] }}"]
+    when: res_def['inbound_nat_rules'] is defined
+
+  - set_fact:
+      inbound_nat_rules: []
+    when: res_def['inbound_nat_rules'] is not defined
+
+  - set_fact:
+      frontend_ip_configurations: ["{{ res_def['frontend_ip_configurations'] }}"]
+    when: res_def['frontend_ip_configurations'] is defined
+
+  - set_fact:
+      frontend_ip_configurations: []
+    when: res_def['frontend_ip_configurations'] is not defined
+
+  - set_fact:
+      backend_address_pools: ["{{ res_def['backend_address_pools'] }}"]
+    when: res_def['backend_address_pools'] is defined
+
+  - set_fact:
+      backend_address_pools: []
+    when: res_def['backend_address_pools'] is not defined
+
+  - set_fact:
+      probes: ["{{ res_def['probes'] }}"]
+    when: res_def['probes'] is defined
+
+  - set_fact:
+      probes: []
+    when: res_def['probes'] is not defined
+
+  - set_fact:
+      inbound_nat_pools: ["{{ res_def['inbound_nat_pools'] }}"]
+    when: res_def['inbound_nat_pools'] is defined
+
+  - set_fact:
+      inbound_nat_pools: []
+    when: res_def['inbound_nat_pools'] is not defined
+
+  - set_fact:
+      load_balancing_rules: ["{{ res_def['load_balancing_rules'] }}"]
+    when: res_def['load_balancing_rules'] is defined
+
+  - set_fact:
+      load_balancing_rules: []
+    when: res_def['load_balancing_rules'] is not defined
+
```

Comment:
Why not use `default` function?

Reply:
Even better, I think there's a `default` tag in the schemas

Reply:
@abraverm  I tried it, for some reason it didn't work. It's still under debugging, the dict and list of ansible are not really good, needed to try different ways to make it work and beautiful. Thanks @14rcole will try it.